pennylane/wires.py

```diff
@@ -412,15 +412,7 @@ def all_wires(list_of_wires, sort=False):
         >>> Wires.all_wires(list_of_wires)
         <Wires = [4, 0, 1, 3, 5]>
         """
-        combined = []
-        seen_labels = set()
-        for wires in list_of_wires:
-            if not isinstance(wires, Wires):
-                raise WireError(f"Expected a Wires object; got {wires} of type {type(wires)}")
-
-            extension = [label for label in wires.labels if label not in seen_labels]
-            combined.extend(extension)
-            seen_labels.update(extension)
+        combined = list(dict.fromkeys(sum([wires.tolist() if isinstance(wires, Wires) else Wires(wires).tolist() for wires in list_of_wires], [])))
```

Comment:
Is there actually much of a performance increase with this change?

This line isn't very readable, so unless it's extremely performance necessary, I'd at least recommend breaking it up into multiple lines and making clear what each component is doing with explanatory variables.

```
to_list = lambda w: w.to_list() is inistance(wires, Wires) else Wires(wires).tolist()
all_wires_in_one_list = sum((to_list(w) for w in list_of_wires), [])
combined = list(dict.fromkeys(combined_to_one_list))
```

Reply:
I did see noticable performance gain in the benchmark from #2430 but you should probably just run this through which ever set of tests you use to evaluate performance impact of changes in the code base.

Reply:
I took a look at this and the speedup is from not running the manual comparisons and extraction loops of the labels, as far as I can tell. It's not so important that it is a one-liner, so I'd suggest
```suggestion
        converted_wires = (wires if isinstance(wires, Wires) else Wires(wires) for wires in list_of_wires)
        all_wires_list = sum((w.tolist() for w in converted_wires), [])
        combined = list(dict.fromkeys(all_wires_list))
```
where we can skip the first line if we want to maintain the previous behaviour, and simply raise an error if the second line crashed with the according attribute error of `tolist` not being present (If a wires object would be of a different type, say a numpy array, that has a `tolist` method, this would silently work. However I do not really see a problem with this :thinking: )
The first line costs some extra time, though.

Reply:
Agree. This version is easier to read and should have the same performance.

Reply:
For some reason I looked at this for so long that I wound up reproducing @albi3ro's version without noticing :facepalm: :D

Reply:
This is implemented now.